clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1012,6 +1011,8 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     DefineTypeSize("__UINTCAP_MAX__", TI.getIntTypeByWidth(CapRange, false), TI, Builder);
 
     if (TI.areAllPointersCapabilities()) {
+      Builder.defineMacro("__CHERI__", "1");
```

Comment:
"1" is the default, might be worth dropping the explicit argument whilst touching this code as it has no real semantic meaning (beyond being non-zero so #if works)?